src/TypeSpec.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -38,22 +40,36 @@ export async function $onEmit(context: EmitContext<AzureNetEmitterOptions>) {
     }
     logger.info("Starting Microsoft Generator Csharp emitter.");
     setSDKContextOptions(azureSDKContextOptions);
-    await $OnMGCEmit(context);
+    const sdkContext = await createSdkContext(
+        context,
+        "@azure-tools/typespec-csharp"
+    );
+    const csharpEmitterContext = { ...sdkContext, logger: logger };
+    const root = createModel(csharpEmitterContext);
+    
     const outputFolder = resolvePath(
         context.emitterOutputDir ?? "./tsp-output"
     );
     const isSrcFolder = path.basename(outputFolder) === "src";
     const generatedFolder = isSrcFolder
         ? resolvePath(outputFolder, "Generated")
         : resolvePath(outputFolder, "src", "Generated");
-    if (options.skipSDKGeneration !== true) {
+    if (root) {
+        if (!fs.existsSync(generatedFolder)) {
+            fs.mkdirSync(generatedFolder, { recursive: true });
+          }
```

Comment:
since we are not using the `fs` from the system to write the file, we called typespec compiler's hook, do we still need to create those directories? I think typespec compiler should be able to handle it

Reply:
not sure I follow this is the generated folder directory. TypeSpec compiler wouldn't create this.